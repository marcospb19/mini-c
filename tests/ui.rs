#![feature(custom_inner_attributes)]
#![rustfmt::skip]

use sushi::test_sushi;
use unindent::unindent;

macro_rules! test {
    ($input:expr) => {
        insta::assert_display_snapshot!(test_sushi($input).unwrap())
    };
}

#[test]
fn empty_statements() {
    test!(unindent("
        ;
        ;;
    "));
}

#[test]
fn empty_program() {
    test!("");
}

#[test]
fn arithmetic_expression() {
    test!("1 + 1 * 2 + 3 / (5 - 3);");
}

#[test]
fn missing_trailing_semi_colon() {
    test!("1 + 1 * 2 + 3 / (5 - 3)");
}

#[test]
fn unfinished_arithmetic_expression() {
    test!("1 -;");
}
