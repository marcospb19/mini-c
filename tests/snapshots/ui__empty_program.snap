---
source: tests/ui.rs
expression: "test_sushi(\"\").unwrap()"
---
### stdout ###
[EMPTY]
### stderr ###
[EMPTY]

